### PR TITLE
feat(nodestore): Implement {get,delete}_multi in Bigtable backend

### PR DESCRIPTION
**Pull request by mattrobenolt**

### CODE REVIEW

**Comment by mattrobenolt** [comment]
There's also a `set_mutli` for completeness, but this isn't used in practice, so not implementing.

File: src/sentry/nodestore/bigtable/backend.py [hunk 1]
```diff
@@ -25,6 +27,32 @@
...
+    except KeyError:
+        # if missing, we acquire our lock to initialize a new one
+        with _connection_lock:
+            try:
+                # It's possible that the lock was blocked waiting
+                # on someone else who already initialized, so
+                # we first check again to make sure this isn't the case.
+                return _connection_cache[key]
+            except KeyError:
+                _connection_cache[key] = (
```
Comment (tkaemming) [inline src/sentry/nodestore/bigtable/backend.py]: I'm assuming you looked to make sure this is actually threadsafe. (I didn't.)

File: src/sentry/nodestore/bigtable/backend.py [hunk 2]
```diff
@@ -168,6 +194,29 @@ def set(self, id, data, ttl=None):
...
+                (
+                    self.thread_pool.submit(self.get, id)
+                    for id in id_list
+                )
+            )
+        }
+
+    def delete_multi(self, id_list):
+        if len(id_list) == 1:
+            return self.delete(id_list[0])
```
Comment (tkaemming) [inline src/sentry/nodestore/bigtable/backend.py]: nit: this sort of implies `self.delete` has a return value that gets discarded on the bulk operation — might make sense to split the `self.delete` and `return` into two separate statements

File: src/sentry/nodestore/bigtable/backend.py [hunk 3]
```diff
@@ -168,6 +194,29 @@ def set(self, id, data, ttl=None):
...
+        if len(id_list) == 1:
+            id = id_list[0]
+            return {id: self.get(id)}
+
+        return {
+            id: future.result()
+            for id, future in izip(
+                id_list,
+                (
+                    self.thread_pool.submit(self.get, id)
```
Comment (tkaemming) [inline src/sentry/nodestore/bigtable/backend.py]: You can use `self.thread_pool.map` if you'd like (you probably know that already)

**Review by tkaemming — APPROVED** [review]

File: src/sentry/nodestore/bigtable/backend.py (hunk 1 above)
Comment (mattrobenolt) [reply src/sentry/nodestore/bigtable/backend.py]: Yeah, it is.

**Pull request merged**